apps/showcase/src/services/webcontainer/webcontainer.service.ts

```diff
@@ -0,0 +1,96 @@
+import {Injectable} from '@angular/core';
+import {FileSystemTree} from '@webcontainer/api';
+import {BehaviorSubject, distinctUntilChanged, map} from 'rxjs';
+import {MonacoTreeElement} from 'ngx-monaco-tree';
+import {convertTreeRec, getFilesTreeFromContainer} from './webcontainer.helpers';
+import {WebContainerRunner} from './webcontainer-runner';
+
+/** List of files or directories to exclude from the file tree */
+const EXCLUDED_FILES_OR_DIRECTORY = ['node_modules', '.angular', '.vscode'];
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WebContainerService {
+  public readonly runner = new WebContainerRunner();
+  private readonly monacoTree = new BehaviorSubject<MonacoTreeElement[]>([]);
+
+  public monacoTree$ = this.monacoTree.pipe(
+    distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b))
+  );
```

Comment:
maybe cleaner to add a `share()` as this  `distinctUntilChanged` is a bit costly